lib/utils/queryable.js

```diff
@@ -141,20 +185,30 @@ const setter = ({ data, key, value, force }) => {
       // appended to the resulting obj is not an array index, then it
       // should throw since we can't append arbitrary props to arrays
       const shouldNotAddPropsToArrays =
+        typeof keys[0] !== 'symbol' &&
         Array.isArray(_data[_key]) &&
         Number.isNaN(Number(keys[0]))
 
       const overrideError =
         haveContents &&
-        (shouldNotOverrideLiteralValue || shouldNotAddPropsToArrays)
-
+        shouldNotOverrideLiteralValue
       if (overrideError) {
         throw Object.assign(
           new Error(`Property ${key} already has a value in place.`),
           { code: 'EOVERRIDEVALUE' }
         )
       }
 
+      const addPropsToArrayError =
+        haveContents &&
+        shouldNotAddPropsToArrays
+      if (addPropsToArrayError) {
+        throw Object.assign(
+          new Error(`Can't add property ${key} to an Array.`),
```

Comment:
what happens when it's an object, but i use `[]`? iow, `foo.bar=3&foo[]=4`?

Reply:
oohhh I really think we should throw an error in that case 🤔

Reply:
what about:

```
$ npm pkg set foo.bar[]=3 foo[]=4
npm ERR! code ENOAPPEND
npm ERR! Can't use append syntax in non-Array element

npm ERR! A complete log of this run can be found in:
npm ERR!     $HOME/.npm/_logs/2021-07-13T21_55_21_310Z-debug.log
```

Reply:
looks great! i guess it's not a good use case to use `[]` syntax on an object that consists entirely of arraylike keys, and convert it to an array :-)

Reply:
let's call that too much of an edge case if you agree 🙃

Reply:
agreed, since you get an error for now we can always change it later